ElrondNetwork/elrond-go
# EN-8070: api request logging middleware

Implemented and integrated a new middleware that will log different data about a request (path, request body, status code, response body, and duration) if it takes longer than a pre-defined threshold **or**it is unsuccessful (not 200 status code).
Introduced a wrapper over the gin's Writer in order to fetch the response.

Testing procedure:
-switch `LoggingEnabled` to `true` in `api.toml`
-make calls to any API endpoint
-if the execution time exceeds the threshold (at the time of opening the PR, the threshold was 100 microseconds so we can test and measure) or it is unsuccessful, you should see a `log.Debug` containing `[...] api request` in node's logs

Example:
-make a call to the balance endpoint. `nodeUrl:port/address/erd1.../balance` . alter the bech32 address so the request is failed with code 400. you should see this logged
